Compilation failure on Windows when using clang-cl

[install-x64-windows-bitwig-rel-out.log](https://github.com/BLAKE3-team/BLAKE3/files/14270326/install-x64-windows-bitwig-rel-out.log)

Maybe easier like this:

```
[7/9] C:\PROGRA~1\MICROS~3\2022\COMMUN~1\VC\Tools\Llvm\x64\bin\clang-cl.exe     /nologo /DWIN32 /D_WINDOWS /W3 /utf-8  /nologo /DWIN32 /D_WINDOWS /utf-8 /w /MP -m64 /MT /O2 /Oi /Gy /DNDEBUG /Z7 /DNDEBUG /O2 /Ob3 /D_DISABLE_STRING_ANNOTATION /D_DISABLE_VECTOR_ANNOTATION /D_ITERATOR_DEBUG_LEVEL=0 /showIncludes /FoCMakeFiles\blake3.dir\blake3_dispatch.c.obj /FdCMakeFiles\blake3.dir\blake3.pdb -c -- C:\j\workspace\ext-libs\src\vcpkg\buildtrees\blake3\src\1.5.0-44da76e0b5.clean\c\blake3_dispatch.c
FAILED: CMakeFiles/blake3.dir/blake3_dispatch.c.obj 
C:\PROGRA~1\MICROS~3\2022\COMMUN~1\VC\Tools\Llvm\x64\bin\clang-cl.exe     /nologo /DWIN32 /D_WINDOWS /W3 /utf-8  /nologo /DWIN32 /D_WINDOWS /utf-8 /w /MP -m64 /MT /O2 /Oi /Gy /DNDEBUG /Z7 /DNDEBUG /O2 /Ob3 /D_DISABLE_STRING_ANNOTATION /D_DISABLE_VECTOR_ANNOTATION /D_ITERATOR_DEBUG_LEVEL=0 /showIncludes /FoCMakeFiles\blake3.dir\blake3_dispatch.c.obj /FdCMakeFiles\blake3.dir\blake3.pdb -c -- C:\j\workspace\ext-libs\src\vcpkg\buildtrees\blake3\src\1.5.0-44da76e0b5.clean\c\blake3_dispatch.c
C:\j\workspace\ext-libs\src\vcpkg\buildtrees\blake3\src\1.5.0-44da76e0b5.clean\c\blake3_dispatch.c(34,23): error: token is not a valid binary operator in a preprocessor subexpression
#elif defined(IS_X86) and defined(_MSC_VER)
      ~~~~~~~~~~~~~~~~^~~
1 error generated.
ninja: build stopped: subcommand failed.
```

--- a/ports/blake3/fix-windows-arm-build-error.patch
+++ b/ports/blake3/fix-windows-arm-build-error.patch
@@ -7,7 +7,7 @@ index af6c3da..dce85b4 100644
  #define ATOMIC_LOAD(x) x
  #define ATOMIC_STORE(x, y) x = y
 -#elif defined(_MSC_VER)
-+#elif defined(IS_X86) and defined(_MSC_VER)
++#elif defined(IS_X86) && defined(_MSC_VER)
  #define ATOMIC_INT LONG
  #define ATOMIC_LOAD(x) InterlockedOr(&x, 0)
  #define ATOMIC_STORE(x, y) InterlockedExchange(&x, y)

--- a/ports/blake3/vcpkg.json
+++ b/ports/blake3/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "blake3",
   "version": "1.5.0",
+  "port-version": 1,
   "description": "BLAKE3 cryptographic hash function.",
   "homepage": "https://github.com/BLAKE3-team/BLAKE3",
   "license": "CC0-1.0 OR Apache-2.0",

--- a/versions/b-/blake3.json
+++ b/versions/b-/blake3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "62d312f1abb1e3acf086050e4e14d650973f5568",
+      "version": "1.5.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "3d62ef88b0bacbb197e9bb921b8b7d0fea336f1d",
       "version": "1.5.0",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -654,7 +654,7 @@
     },
     "blake3": {
       "baseline": "1.5.0",
-      "port-version": 0
+      "port-version": 1
     },
     "blas": {
       "baseline": "2023-03-25",

